### PR TITLE
Forward divide by zero exception code to kernel

### PR DESCRIPTION
**Pull request by vtikoo**
Signed-off-by: Vikas Tikoo <vitikoo@microsoft.com>

### CODE REVIEW

File: tools/myst/enc/enc.c
Comment (jxyang) [inline tools/myst/enc/enc.c:145]: The exit info doesn't indicate whether this is a integer divide-by-0 or floating point divide-by-0. I think `FPE_INTDIV` is good enough here. Please update the comment.

**Review by jxyang — APPROVED** [review]

**Comment by jxyang** [comment]
For the failed test, we will need Paul's pseudo fork PR to forward the signals on Linux target.

File: tools/myst/enc/enc.c
Comment (vtikoo) [reply tools/myst/enc/enc.c:145]: Thanks for clarifying. What about other SIGFPE cases like overflow, inexact result, out of range etc., will they generate a different `oe_exception_code`?

**Comment by vtikoo** [comment]
> For the failed test, we will need Paul's pseudo fork PR to forward the signals on Linux target.

Okay, I'll hold off until #414 is merged.

File: tools/myst/enc/enc.c
Comment (bodzhang) [reply tools/myst/enc/enc.c:146]: OE code maps x86 #DE (Divide Error Exception) to `OE_EXCEPTION_DIVIDE_BY_ZERO`. #DE is integer divide-by-zero exception. Floating point divided-by-zero exception is part of x86 #MF (`OE_EXCEPTION_X87_FLOAT_POINT`) or #XM (`OE_EXCEPTION_SIMD_FLOAT_POINT`). For accurate [si_code](https://www.man7.org/linux/man-pages/man0/signal.h.0p.html) for SIGFPE, the implementation likely needs to parse the FPU status register or MXCSR register (for SIMD). Please take this opportunity to add "ATTN" comments to the code. 

ATTN: Consider implementing accurate si-code for OE_EXCEPTION_X87_FLOAT_POINT

ATTN: Consider forwarding OE_EXCEPTION_BOUND_OUT_OF_RANGE, OE_EXCEPTION_ACCESS_VIOLATION, OE_EXCEPTION_MISALIGNMENT, OE_EXCEPTION_SIMD_FLOAT_POINT as signal.

Comment (vtikoo) [reply tools/myst/enc/enc.c:146]: > ATTN: Consider implementing accurate si-code for OE_EXCEPTION_X87_FLOAT_POINT

Can we replace the si_code to FPE_FLTDIV?

Comment (vtikoo) [reply tools/myst/enc/enc.c:146]: > Please take this opportunity to add "ATTN" comments to the code.

Added.

Comment (jxyang) [reply tools/myst/enc/enc.c:146]: @bodzhang Yes we should fill in the accurate `si_code` as long as OE provides the details.

Comment (mingweishih) [reply tools/myst/enc/enc.c:146]: @vtikoo 
`FPE_FLTINV`: Invalid floating-point operation
may be a more general option to put for now.
As @bodzhang mentions, we need some work to determine the accurate code, which is captured by the ATTN comment.

Comment (bodzhang) [reply tools/myst/enc/enc.c:146]: I think for OE_EXCEPTION_DIVIDE_BY_ZERO, FPE_INTDIV is the right si_code.

Comment (mingweishih) [reply tools/myst/enc/enc.c:146]: Yes, I was referring to the OE_EXCEPTION_X87_FLOAT_POINT case

File: tools/myst/enc/enc.c
Comment (mingweishih) [inline tools/myst/enc/enc.c:136]: I would suggest still keep the following comment so that we can improve this later.

ATTN: Consider implementing accurate si-code for OE_EXCEPTION_X87_FLOAT_POINT

Comment (vtikoo) [reply tools/myst/enc/enc.c:138]: Makes sense, added the comment back.

**Review by mingweishih — APPROVED** [review]

**Pull request merged**
